openvdb_points_clarisse/clarisse/module/Resource_OpenVDB_Points.cc

```diff
@@ -0,0 +1,489 @@
+
+            array[arrayIndex][0] = positionWorldSpace[0];
+            array[arrayIndex][1] = positionWorldSpace[1];
+            array[arrayIndex][2] = positionWorldSpace[2];
+
+            arrayIndex++;
+        }
+    }
+
+    convert_progress_bar->destroy();
+
+    // construct the point cloud
+
+    AppProgressBar* cloud_progress_bar = application.create_progress_bar(CoreString("Creating Clarisse Point Cloud"));
+
+    GeometryPointCloud pointCloud;
+
+    pointCloud.init(array.get_count());
+    pointCloud.init_positions(array.get_data());
+
+    cloud_progress_bar->destroy();
+
+    // load the velocities (only if v attribute exists)
+
+    if (loadVelocities && tree.cbeginLeaf()->hasAttribute("v"))
+    {
+        AppProgressBar* velocity_progress_bar = application.create_progress_bar(CoreString("Loading Velocities into Point Cloud"));
+
+        arrayIndex = 0;
+
+        for (PointDataTree::LeafCIter leaf = tree.cbeginLeaf(); leaf; ++leaf)
+        {
+            const AttributeHandle<openvdb::Vec3f>::Ptr velocityHandle =
+                AttributeHandle<openvdb::Vec3f>::create(leaf->constAttributeArray("v"));
+
+            for (PointDataTree::LeafNodeType::IndexOnIter iter = leaf->beginIndexOn(); iter; ++iter)
+            {
+                const openvdb::Vec3f velocity = velocityHandle->get(*iter);
+
+                array[arrayIndex][0] = velocity.x();
+                array[arrayIndex][1] = velocity.y();
+                array[arrayIndex][2] = velocity.z();
+
+                arrayIndex++;
+            }
+        }
+
+        pointCloud.init_velocities(array.get_data());
+
+        velocity_progress_bar->destroy();
+    }
+
+    return new ParticleCloud(pointCloud);
+}
+
+
+GeometryPointPropertyCollection*
+create_clarisse_particle_cloud_geometry_property(OfApp& application, ResourceData_OpenVDBPoints& data)
+{
+    using resource_internal::PropertyDataInfo;
+    using resource_internal::buildProperty;
+
+    GeometryPointPropertyCollection* collection = new GeometryPointPropertyCollection;
+
+    openvdb::tools::PointDataGrid::Ptr grid = data.grid();
+
+    if(!grid)  return collection;
+
+    const openvdb::tools::PointDataTree& tree = grid->tree();
+    const openvdb::tools::PointDataTree::LeafCIter iter = tree.cbeginLeaf();
+
+    if(!iter) return collection;
+
+    const unsigned int numPoints = pointCount(tree);
+
+    if(numPoints == 0) return collection;
+
+    CoreVector<GeometryPointProperty*> properties;
+
+    const openvdb::tools::AttributeSet::Descriptor& descriptor = iter->attributeSet().descriptor();
+
+    for (openvdb::tools::AttributeSet::Descriptor::ConstIterator it = descriptor.map().begin(),
+        end = descriptor.map().end(); it != end; ++it)
+    {
+        const openvdb::Name name = it->first;
+        const openvdb::Name& valueType = descriptor.valueType(it->second);
+        ResourceProperty::Type resourceType(ResourceProperty::TYPE_COUNT);
+
```

Comment:
Forgot to push up one minor fix here:

   // if this is a group, skip
   if(descriptor.hasGroup(name)) continue;